plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -1557,6 +1558,18 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
             nullListDenotation
             (runCostingFunOneArgument . paramNullList)
 
+    toBuiltinMeaning _semvar DropList =
+        let dropListDenotation :: Int -> SomeConstant uni [a] -> BuiltinResult (Opaque val [a])
```

Comment:
The time taken for this will probably be linear in the value (not the size!) of the first argument.  That'll mean susing some newtype wrapper, like [this](https://github.com/IntersectMBO/plutus/blob/983eba57bb6c2240780d078fa4618307d96f72a8/plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs#L1998) .  We already have [IntegerCostedLiterally](https://github.com/IntersectMBO/plutus/blob/983eba57bb6c2240780d078fa4618307d96f72a8/plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/ExMemoryUsage.hs#L192), but this has an `Int` so that may need another wrapper.

Reply:
> The time taken for this will probably be linear in the value (not the size!) of the first argument. 

No, I think it's gonna be a minimum of

- linear in the value of the first argument
- linear in the length of the second argument

since `drop maxBound []` equals `[]`.

Reply:
> Actually, why did I even make it `Int`, what's the point? Looks like a thinko.

It's proper `Integer` in there now.

Reply:
> No, I think it's gonna be a minimum of
>    linear in the value of the first argument
>    linear in the length of the second argument
> since drop maxBound [] equals [].

I deliberately ignored that case!  If we include the length of the list in the cost calculation then we'll have to traverse the etire list to measure the length when we're about to calculate the cost, and that'll slow things down.  If we just base it on the number of things we want to drop then we'll still get a conservative estimate of the cost, and it'll be faster.  Also, if you drop the whole of the list then you've probably done something wrong, and people probably won't submit scripts where that actually happens (or at least not often).

Reply:
Oh I see. OK, that's clever. Yeah, I think I agree, although I'd perhaps prefer to store lists together with their length, although that'd perhaps slow everything else down. Fair enough!

@ramsay-t ignore what I said and only consider the original comment in this thread.

Reply:
> I'd perhaps prefer to store lists together with their length

I've wondered about doing that in the past (and maybe something similar with Data for instance), but what happens if someone lies about the length of a list in their script?  Maybe that could be checked during deserialisation ...